sw/device/tests/mbx_smoketest.c

```diff
@@ -0,0 +1,666 @@
+      break;
+    }
+    case kTopDarjeelingPlicPeripheralMbx1: {
+      plic_periph_base_irq_id = kTopDarjeelingPlicIrqIdMbx1MbxReady;
+      break;
+    }
+    case kTopDarjeelingPlicPeripheralMbx2: {
+      plic_periph_base_irq_id = kTopDarjeelingPlicIrqIdMbx2MbxReady;
+      break;
+    }
+    case kTopDarjeelingPlicPeripheralMbx3: {
+      plic_periph_base_irq_id = kTopDarjeelingPlicIrqIdMbx3MbxReady;
+      break;
+    }
+    case kTopDarjeelingPlicPeripheralMbx4: {
+      plic_periph_base_irq_id = kTopDarjeelingPlicIrqIdMbx4MbxReady;
+      break;
+    }
+    case kTopDarjeelingPlicPeripheralMbx5: {
+      plic_periph_base_irq_id = kTopDarjeelingPlicIrqIdMbx5MbxReady;
+      break;
+    }
+    case kTopDarjeelingPlicPeripheralMbx6: {
+      plic_periph_base_irq_id = kTopDarjeelingPlicIrqIdMbx6MbxReady;
+      break;
+    }
+    case kTopDarjeelingPlicPeripheralMbxJtag: {
+      plic_periph_base_irq_id = kTopDarjeelingPlicIrqIdMbxJtagMbxReady;
+      break;
+    }
+    case kTopDarjeelingPlicPeripheralMbxPcie0: {
+      plic_periph_base_irq_id = kTopDarjeelingPlicIrqIdMbxPcie0MbxReady;
+      break;
+    }
+    case kTopDarjeelingPlicPeripheralMbxPcie1: {
+      plic_periph_base_irq_id = kTopDarjeelingPlicIrqIdMbxPcie1MbxReady;
+      break;
+    }
+    default: {
+      CHECK(false,
+            "get_lowest_irq() called for a plic_irq_id that is not a mailbox!");
+      return 0;
+    }
+  }  // switch(peripheral)
+
+  return plic_periph_base_irq_id;
+}
+
+////////////////////////
+// CONFIGURE MEMORIES //
+////////////////////////
+
+enum {
+  kSramStart = TOP_DARJEELING_SRAM_CTRL_MBOX_RAM_BASE_ADDR,
+  kSramEnd = TOP_DARJEELING_SRAM_CTRL_MBOX_RAM_BASE_ADDR +
+             TOP_DARJEELING_SRAM_CTRL_MBOX_RAM_SIZE_BYTES,
+  kMbxSizeDWORDS = 8,  // The size we are allocating to each mbx for this test
+                       // (imbx + ombx == kMbxSizeDWORDS * 2)
+};
+
+static_assert(
+    NUM_MBX * (kMbxSizeDWORDS * 2) <=
+        (kSramEnd - kSramStart) / sizeof(uint32_t),
+    "As specified, the mailbox memories cannot fit in the backing SRAM!");
+
+// Backing storage for objects used by the mailbox handler(s)
+// (dif_mbx_transaction_t)
+static uint32_t gDataDWORDS[NUM_MBX][kMbxSizeDWORDS];
+
+/**
+ * Setup the mailbox CSRs
+ *
+ * - Setup imbx/ombx base+limit addresses to match SoC memory
+ * - Other misc init tasks (addr_range_valid, etc)
+ */
+void configure_mbx_peripherals(void) {
+  uint32_t mbx_size_bytes = kMbxSizeDWORDS * sizeof(uint32_t);
+
+  for (size_t i = 0; i < NUM_MBX; ++i) {
+    uint32_t mbx_region_base = kSramStart + (mbx_size_bytes * 2 * i);
+    // Set the memory ranges
+    dif_mbx_range_config_t config = {
+        .imbx_base_addr = mbx_region_base,
+        .imbx_limit_addr =
+            mbx_region_base + mbx_size_bytes,  // imbx_limit == ombx_base
+        .ombx_base_addr = mbx_region_base + mbx_size_bytes,
+        .ombx_limit_addr = mbx_region_base + (mbx_size_bytes * 2),
```

Comment:
The 'limit_addr' values of the IP block are actually _inclusive_, ie. the registers are programmed with the start address of the final DWORD within the valid address range, although the documentation still requires modifying to make that explicit. The DIF also needs updating accordingly.
Perhaps you could change the values calculated here, and I'll raise a PR to change the DIF, documentation and anything else I find that needs updating.

Reply:
https://github.com/lowRISC/opentitan/pull/20774
Thanks for opening it!